src/core/alert/AlertProvider.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { createContext, useContext, useState } from "react";
+
+import { Alert } from "@components"
+
+import { AlertsContainer } from "./style"
+
+type Alert = {
+    msg: string,
+    id: number,
+    variant: boolean
+}
+
+interface Props {
+    alerts: Alert[],
+    addAlert: (msg: string) => void
+}
+
+const STATE: Props = {
+    alerts: [],
+    addAlert: () => { }
+}
+
+const Context = createContext(STATE);
+
+export const AlertProvider = (props: any) => {
+    const [alerts, setAlerts] = useState<Alert[]>([]);
+
+    // const sleep = (ms: number) => new Promise(resolve => setTimeout(resolve, ms));
+
+    const addAlert = (msg: string) => {
+        const id = alerts.length > 0 ? alerts[alerts.length - 1].id + 1 : 0;
+        setAlerts([...alerts, { msg: msg, id: id, variant: false }]);
+    }
+
+    const removeAlert = async (id: number) => {
```

Comment:
index as arg and simply remove this element from array

Reply:
same as above